lib/logic/box_communicator.dart

```diff
@@ -32,46 +72,55 @@ class BoxCommunicator {
     //storage box stores the number of received bytes at 'totalSizeInBytes'
     Box storage = await Hive.openBox('storage');
     //opens the box for the current connected Sensorbox
-    Box box = await Hive.openBox(boxName);
+    LazyBox box = await Hive.openLazyBox(boxName);
+    Box boxes = await Hive.openBox('boxes');
+    boxes.add(boxName);
 
     //if no data has been received: start with 0.
     int totalSizeInBytes = storage.get('totalSizeInBytes', defaultValue: 0);
-
-    while (true) {
-      //is the data limit reached?
-      if (totalSizeInBytes > dataLimitInkB * 1000) {
-        print("data limit reached");
-        return;
-      }
-      //make getData call to collect data chunks or files from box
-      final response = await retry(
-          () => http.get(boxIP + "/api/getData").timeout(Duration(seconds: 3)),
-          retryIf: (e) => e is SocketException || e is TimeoutException);
-
-      if (response.statusCode == 200) {
-        var id = jsonDecode(response.body)["_id"];
-        if (id == null) {
-          print('response had no id');
-        }
-        //all files of the current box have been downloaded
-        if (box.get(id) != null) {
-          print("got all files");
+    try {
+      while (true) {
+        //is the data limit reached?
+        if (totalSizeInBytes > dataLimitInkB * 1000) {
+          print("data limit reached");
           break;
         }
-        box.put(id, response.body);
-        //add size of this response to size of all data combined (for limiting data usage)
-        storage.put(
-            'totalSizeInBytes', totalSizeInBytes + response.contentLength);
-        print("new entry at: ");
-        print(DateTime.now());
-      } else {
-        // If the server did not return a 200 OK response,
-        // then throw an exception.
-        print(response.statusCode);
-        print('failed to get any data');
+        //make getData call to collect data chunks or files from box
+        final response = await r.retry(
+            () =>
+                http.get(boxIP + "/api/getData").timeout(Duration(seconds: 3)),
+            retryIf: (e) => e is SocketException || e is TimeoutException);
+
+        if (response.statusCode == 200) {
+          var id = jsonDecode(response.body)["_id"];
+          if (id == null) {
+            print('response had no id');
```

Comment:
what happens if there is no id? id continues to be used in the next lines, maybe omit (use else block)?